### ISSUE TITLE
 TypeError: Cannot read property 'path' of undefined

### ISSUE THREAD

**Issue by gutenye** [gatsbyjs/gatsby#1050]
```
TypeError: Cannot read property 'path' of undefined
      at resolve (node_modules/gatsby/dist/schema/infer-graphql-type.js:323:23)
```

Give this query

```
query HomePage {
  allPostsJson {
    edges {
      node {
        image {
         relativePath
       }
     }
   }
}
```

`image` is a `File` type.

The source code for `node_modules/gatsby/dist/schema/infer-graphql-type.js:323` is

```javascript
function inferFromUri(key, types) {
  var fileField = types.find(function (type) {
    return type.name === `File`;
  });

  if (!fileField) return;

  return {
    type: fileField.nodeObjectType,
    resolve: function resolve(node, a, _ref5) {
      var path = _ref5.path;            // Line 323 HERE
```

Run in `pages/index.js` works without any error. For context is provided.
Run in `createPages() in gatsby-node.js` got this error. For context is undefined.

Seems like context is not provided when run inside `createPages() in gatsby-node.js`.

**Comment by KyleAMathews** [comment]
You want to take a crack at solving this? Looks like you've figured out most of the bug already :-) I'm working on a tutorial right now for the beta launch so won't get to this for a few days.

**Comment by gutenye** [comment]
Turns out you need to provide context by yourself.

```
graphql(`query HomePage {
  allPostsJson {
    edges {
      node {
        image {
         relativePath
       }
     }
   }
}`, { path: '/' })
```

Is this consider a bug or by design?

**Comment by KyleAMathews** [comment]
The bug is that there's parts of the code that try to use the context as for most graphql queries there's a context automatically provided (e.g. the "path" for the current graphql query).

Providing your own context actually could introduce a bug as there's code paths which infer things from the path. Since your "path" is arbitrarily chosen it could cause subtle bugs.

The context almost never should be set. It's only there because it's used internally. I could also see it being used as an escape hatch if someone wanted to add something custom to the graphql schema and wanted to use some custom variables on the context as part of that. But that seems unlikely

I fixed this in #1050
